lnbits/settings.py

```diff
@@ -1,92 +1,298 @@
+    lnpay_api_endpoint: Optional[str] = Field(default=None)
+    lnpay_api_key: Optional[str] = Field(default=None)
+    lnpay_wallet_key: Optional[str] = Field(default=None)
+    lntxbot_api_endpoint: Optional[str] = Field(default=None)
+    lntxbot_key: Optional[str] = Field(default=None)
+    opennode_api_endpoint: Optional[str] = Field(default=None)
+    opennode_key: Optional[str] = Field(default=None)
+    spark_url: Optional[str] = Field(default=None)
+    spark_token: Optional[str] = Field(default=None)
+    lntips_api_endpoint: Optional[str] = Field(default=None)
+    lntips_api_key: Optional[str] = Field(default=None)
+    lntips_admin_key: Optional[str] = Field(default=None)
+    lntips_invoice_key: Optional[str] = Field(default=None)
+
+    # boltz
+    boltz_network: str = Field(default="main")
+    boltz_url: str = Field(default="https://boltz.exchange/api")
+    boltz_mempool_space_url: str = Field(default="https://mempool.space")
+    boltz_mempool_space_url_ws: str = Field(default="wss://mempool.space")
+
+    @validator(
+        "lnbits_admin_users",
+        "lnbits_allowed_users",
+        "lnbits_theme_options",
+        "lnbits_admin_extensions",
+        "lnbits_disabled_extensions",
+        "lnbits_allowed_funding_sources",
+        pre=True,
+    )
+    def validate(cls, val):
+        if type(val) == str:
+            val = val.split(",") if val else []
+        return val
+
+    @classmethod
+    def from_row(cls, row: Row) -> "Settings":
+        data = dict(row)
+        return cls(**data)
+
+    class Config:
+        env_file = ".env"
+        env_file_encoding = "utf-8"
+        case_sensitive = False
+        json_loads = list_parse_fallback
+
+
+settings = Settings()
+
+settings.lnbits_path = str(path.dirname(path.realpath(__file__)))
 
 try:
-    LNBITS_COMMIT = (
+    settings.lnbits_commit = (
         subprocess.check_output(
-            ["git", "-C", LNBITS_PATH, "rev-parse", "HEAD"], stderr=subprocess.DEVNULL
+            ["git", "-C", settings.lnbits_path, "rev-parse", "HEAD"],
+            stderr=subprocess.DEVNULL,
         )
         .strip()
         .decode("ascii")
     )
 except:
-    LNBITS_COMMIT = "unknown"
+    settings.lnbits_commit = "docker"
+
+
+if not settings.lnbits_admin_ui:
+    logger.debug(f"Enviroment Settings:")
+    for key, value in settings.dict(exclude_none=True).items():
+        logger.debug(f"{key}: {value}")
+
+
+def set_cli_settings(**kwargs):
+    for key, value in kwargs.items():
+        setattr(settings, key, value)
+
+
+async def check_admin_settings():
+    if settings.lnbits_admin_ui:
+        try:
+            ext_db = importlib.import_module(f"lnbits.extensions.admin").db
+        except:
+            logger.error("could not import module lnbits.extensions.admin database")
+            raise
+
+        async with ext_db.connect() as db:
+            try:
+                row = await db.fetchone("SELECT * FROM admin.settings")
```

Comment:
`admin.settings` CRUD operations should be part of `crud.py` in the `admin` extension (not of core)